addons/website_hr_recruitment/static/src/js/website_hr_applicant_form.js

```diff
@@ -0,0 +1,146 @@
+            $('#phone-message').removeClass('alert-warning').hide();
+            return;
+        }
+
+        /* This regex matches a general type of phone number that may or may
+         * not include country code, area code, and extension. 
+         * It allows for various separators such as spaces, dots, slashes,
+         * and hyphens.
+         * Here some exemple of valid phone numbers:
+         * +1 (555) 555-5555
+         * 555-555-5555
+         * 555.555.5555
+         * 555/555-5555
+         * (555) 555-5555
+         * +44 20 1234 5678
+         * 12345 67890
+         * [Xs(sXs)]Xs[s(sXs)]XsXsXs[sXsXs]Xs[sXsXs]Xs
+         * */
+        const phone_number_regex = /^(?:\+?\(?\d{1,3}\)?[\s./-]?)?(?:\(\d{1,5}\)|\d{1,5})[\s./-]?\d{1,10}[\s./-]?\d{1,10}[\s./-]?\d{1,10}$/;
+
+        if (!phone_number_regex.test(phone_number)) {
+
+            $('#phone-message').removeClass('alert-warning').hide();
+            $(ev.currentTarget).addClass('border-warning');
+            $('#phone-message').text("Your phone number doesn't match any known standard format. If it's not a mistake you can continue.").addClass('alert-warning').show();
+
+        } else {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#phone-message').removeClass('alert-warning').hide();
+        }
+
+    },
+
+    async _onFocusOutLinkedin (ev) {
+
+        const linkedin = $(ev.currentTarget).val();
+
+        if (!linkedin) {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#linkedin-message').removeClass('alert-warning').hide();
+            return;
+        }
+        
+        const linkedin_regex = /^(https?:\/\/)?([\w\.]*)linkedin\.com\/in\/(.*?)(\/.*)?$/;
+
+        if (!linkedin_regex.test(linkedin)) {
+
+            $('#linkedin-message').removeClass('alert-warning').hide();
+            $(ev.currentTarget).addClass('border-warning');
+            $('#linkedin-message').text("The profile that you gave us doesn't seems like a linkedin profile").addClass('alert-warning').show();
+
+        } else {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#linkedin-message').removeClass('alert-warning').hide();
+        }
+
+    },
+
+    async _onFocusOutMail (ev) {
+        const email = $(ev.currentTarget).val();
+
+        if (!email) {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#email-message').removeClass('alert-warning').hide();
+            return;
+        }
+        
+        const job_id = $('#recruitment7').val();
+        const csrf_token = $('#csrf_token').val();
+        this._rpc({
+            route: '/hr/check_recent_application?' + $.param({
+                email: email,
+                job_id: job_id,
+                csrf_token: csrf_token,
+            }),
+        });
+        const response = await fetch(`/hr/check_recent_application?csrf_token=${csrf_token}&email=${email}&job_id=${job_id}`);
```

Comment:
Be careful in general when encoding your urls because it can be prone to injections, here for instance if i input "0h3n-rijaune@yahoo.example.com&job_id=413" in the form, i will inject job_id=413 in the request and my application might go through, so instead let's use the proper rpc call above, using the new rewritten controller method : 

```javascript
const data = await this._rpc({
    route: '/website_hr_recruitment/check_recent_application',
    params: {
        email: email,
        job_id: job_id,
    }
});
```